inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,156 @@
+[Resources](#resources) at the end of this document.
+
+## Prerequisites
+
+Before you do a pull request, you should always file an issue and make sure
+someone from the tidyverse team agrees that it’s a problem, and is happy with
+your basic proposal for fixing it. If you’ve found a bug, first create a minimal
+[reprex](https://www.tidyverse.org/help/#reprex).
+
+## Package mechanics
+
+### Fork, clone, branch
+
+The first thing you'll need to do is to fork the 
+[{{{package}}} GitHub repo](https://github.com/tidyverse/{{{package}}}), and 
+then clone it locally. We recommend that you create a branch for each PR.
+
+### Establish a baseline
+
+Before changing anything, make sure the package still passes `R CMD check`
+locally for you.
+
+``` r
+# install.packages("devtools")
+devtools::check()
+```
+
+### Style
+
+Match the existing code style. This means you should follow the tidyverse
+style guide (see [Resources](#resources)).
+You can also use the [styler](https://CRAN.R-project.org/package=styler) package
+to find code which does not adhere to the style guide.
+
+``` r
+# install.packages("styler")
+styler::style_pkg()
+```
+
+Be careful to only make style changes to the code you are contributing. If you
+find that there is a lot of code that doesn't meet the style guide, it would be
+better to file an issue or a separate PR to fix that first. You can use one of
+styler's other [`style_text`](http://styler.r-lib.org/index.html) variants to
+limit the scope of your styling.
+
+### Documentation
+
+We use [roxygen2](https://cran.r-project.org/package=roxygen2),
+specifically with the [Markdown
+syntax](https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html),
+to create `NAMESPACE` and all `.Rd` files. All edits to documentation
+should be done in roxygen comments above the associated function or
+object.
+
+``` r
+# install.packages("devtools")
+devtools::document()
+```
+
+See the `RoxygenNote` in [DESCRIPTION](DESCRIPTION) for the version of
+roxygen2 being used. 
+
+### Testing
+
+We use [testthat](https://cran.r-project.org/package=testthat). Contributions
+with test cases are easier to accept. If you are not sure what parts of your
+code is covered by tests, run the following to get a local coverage report of
+the package so you can see exactly what lines are not covered in the project.
+
+``` r
+# install.packages("devtools")
+devtools::test_coverage()
+```
+
+Before submitting your changes, make sure that the package either still
+passes `R CMD check`, or that the warnings and/or notes have not _changed_
+as a result of your edits.
+
+### NEWS
+
+Add a bullet to `NEWS.md` with a concise description of the change, if it’s
+something a user would want to know when updating the package. The sentence
+format should include your GitHub username, and links to relevant issue(s)/
+PR(s). We will handle any organization into sub-sections just prior to a
+release.
+
+What merits a bullet?
```

Comment:
I think incorporate this into the first sentence. 

For user-facing changes, add a bullet to `NEWS.md` that concisely describes the change. Small tweaks to the documentation do not need a bullet.